stacker_blueprints/iam_roles.py

```diff
@@ -36,21 +52,60 @@ def __init__(self, *args, **kwargs):
 
     def create_role(self, name, assumerole_policy):
         t = self.template
+        v = self.get_variables()
+
+        role_kwargs = {
+            'AssumeRolePolicyDocument': assumerole_policy,
+        }
+
+        attached_policies = v['AttachedPolicies']
+        if attached_policies and len(attached_policies):
```

Comment:
You shouldn't need the `len` test here, since anything that's length 0 (a empty list, string, dictionary) will evaluate as False in python.  You could replace these two lines with:

```
role_kwargs["ManagedPolicyArns"] = attached_policies or NoValue
```

(You'll want to import `NoValue` from troposphere at the top of the file if it's not already)